### DISCUSSION TITLE
Indexing new documents without indexing the full collection

### DISCUSSION THREAD

**Discussion by Fatima-Haouari** [castorini/pyserini#915] [Q&A] [answered]
Hi,
I used pyserini to index a collection of documents. If my collection of documents was expanded, how can I index just the new documents without indexing the full collection again?

**Comment by lintool** [comment, marked as answer]
If you've modified the collection, including expansion, you'll have to re-index the entire collection.

**Reply by Fatima-Haouari** [reply]
Thanks a lot for quick response. If adding new documents to the collection, do I need re-index the entire collection as well?

**Reply by lintool** [reply]
In Pyserini, yes. For Lucene in general, no. Unfortunately, we have not exposed the hooks to Lucene that allow you to add new documents to the index.

**Reply by Fatima-Haouari** [reply]
Thanks for your help.
